database/migrations/2023_09_12_162426_create_products_table.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+
+    public function up(): void
+    {
+        Schema::create('products', function (Blueprint $table) {
+            $table->uuid('id')->primary();
+            $table->string('name')->unique();
+            $table->foreignUuid('brand_id')->nullable()->constrained();
+            $table->foreignUuid('color_id')->nullable()->constrained();
+            $table->foreignUuid('size_id')->nullable()->constrained();
```

Comment:
brand_id is compulsory
color_id & size_id is nullable then set validation rule & html required accordingly.